contracts/src/main/kotlin/com/contractsAndStates/states/GameBoardState.kt

```diff
@@ -110,8 +119,53 @@ data class AccessPoint(val hexTileIndex: HexTileIndex, val hexTileCoordinate: Li
 data class HexTileIndex(val value: Int) {
 
     init {
-        require(0 <= value && value < GameBoardState.TILE_COUNT) {
-            "value cannot be $value"
-        }
+        require(0 <= value && value < GameBoardState.TILE_COUNT) { "value cannot be $value" }
+    }
+}
+
+/**
+ * Applies to roads.
+ */
+data class TileSideIndex(val value: Int) {
+
+    init {
+        require(0 <= value && value < HexTile.SIDE_COUNT) { "value cannot be $value" }
     }
-}
\ No newline at end of file
+
+    // TODO does + 3 % 5 work?
+    fun opposite() = TileSideIndex(
+            if (value + HexTile.SIDE_COUNT / 2 < HexTile.SIDE_COUNT) value + HexTile.SIDE_COUNT / 2
+            else value - HexTile.SIDE_COUNT / 2)
+
+    fun previous() = TileSideIndex(
+            if (value - 1 < 0) HexTile.SIDE_COUNT - 1
+            else value - 1)
+
+    fun next() = TileSideIndex(
+            if (value + 1 < HexTile.SIDE_COUNT) value + 1
+            else 0)
+}
+
+/**
+ * Applies to settlements.
+ * TODO Confirm that corner 1 is between side 0 and side 1.
```

Comment:
@rogersanick A question for you.

Reply:
Hey @xavierlepretre - you are correct.

Corner 0 is the top-most point of a hextile (between sides 5 and 0)
Corner 1 is the point immediately to the right (one jump clockwise and between sides 0 and 1)